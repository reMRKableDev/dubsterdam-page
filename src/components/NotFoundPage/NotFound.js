import React from "react";
import { StyledDiv, StyledH1 } from "./NotFoundStyles";
import { Link } from "react-router-dom";

const NotFound = () => {
  return (
    <StyledDiv>
      <p>Sorry,</p>
      <StyledH1>
        This Page <span>Doesn't Exist!</span>
      </StyledH1>
      <Link to="/">Back to the Homepage</Link>
    </StyledDiv>
  );
};

export default NotFound;
