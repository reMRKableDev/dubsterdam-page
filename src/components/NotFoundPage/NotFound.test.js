import React from "react";
import { shallow, render } from "enzyme";
import { MemoryRouter } from "react-router-dom";

import NotFound from "./NotFound";

import {
  validateMatchSnapshot,
  validateToEqual,
} from "../../utils/testValidators";

describe("Not Found unit tests", () => {
  it("should validate match snapshot for Not Found component", () => {
    const wrapper = shallow(<NotFound />);

    validateMatchSnapshot(wrapper);
  });

  it("should validate <p> to equal 'Sorry,'", () => {
    const wrapper = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    );
    const notFoundText = wrapper.find("p").text();

    validateToEqual(notFoundText, "Sorry,");
  });

  it("should validate <h1> to equal 'This Page Doesn't Exist!,'", () => {
    const wrapper = render(
      <MemoryRouter>
        <NotFound />
      </MemoryRouter>
    );
    const notFoundText = wrapper.find("h1").text();

    validateToEqual(notFoundText, "This Page Doesn't Exist!");
  });
});
