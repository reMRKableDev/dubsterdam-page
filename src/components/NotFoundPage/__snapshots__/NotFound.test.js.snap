// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found unit tests should validate match snapshot for Not Found component 1`] = `
<styled.div>
  <p>
    Sorry,
  </p>
  <styled.h1>
    This Page 
    <span>
      Doesn't Exist!
    </span>
  </styled.h1>
  <Link
    to="/"
  >
    Back to the Homepage
  </Link>
</styled.div>
`;
